lib/Cake/Test/Case/Model/Datasource/Database/SqlserverTest.php

```diff
@@ -707,4 +707,117 @@ public function testStoredProcedureReturn() {
 		$this->assertEquals(2, $result['value']);
 	}
 
+/**
+ * Test build statement
+ *
+ * @return void
+ */
+	public function testBuildStatement() {
+		$db = $this->getMock('SqlserverTestDb', array('getVersion'), array($this->Dbo->config));
+
+		$db->expects($this->any())
+			->method('getVersion')
+			->will($this->returnValue('11.00.0000'));
+
+		// HAVING
+		$query = array(
+			'fields' => array('user_id', 'COUNT(*) AS count'),
+			'table' => 'articles',
+			'alias' => 'Article',
+			'group' => 'user_id',
+			'order' => array('COUNT(*)' => 'DESC'),
+			'limit' => 5,
+			'having' => array('COUNT(*) >' => 10),
+		);
```

Comment:
Do you mean I should separate `testBuildStatement` into two methods - `testBuildStatementWithHaving()` and `testBuildStatementWithLockingHint()` ?